packages/target-ethers-v5/src/codegen/functions.ts

```diff
@@ -29,11 +30,15 @@ function generateFunction(options: GenerateFunctionOptions, fn: FunctionDeclarat
     !isConstant(fn) && !isConstantFn(fn)
       ? `overrides?: ${isPayable(fn) ? 'PayableOverrides' : 'Overrides'}`
       : 'overrides?: CallOverrides'
-  }): Promise<${
-    options.isStaticCall || fn.stateMutability === 'pure' || fn.stateMutability === 'view'
-      ? generateOutputTypes(!!options.returnResultObject, fn.outputs)
-      : 'ContractTransaction'
-  }>;
+  }): ${
+    options.overrideOutput
+      ? options.overrideOutput
+      : `Promise<${
```

Comment:
Nice refactoring `estimateGas` and `populateTransaction` generator functions with `overrideOutput` option. 

Just a quick review, I think `x ? x : y` can be replaced with `x ?? y`.

Reply:
Added suggestion to commit if so

Reply:
Heh - should have a github action that does the prettier so that wouldn't fail next time :)